types/jsonwebtoken/index.d.ts

```diff
@@ -178,6 +178,11 @@ export function sign(
     secretOrPrivateKey: Secret,
     options?: SignOptions,
 ): string;
+export function sign(
+    payload: string | Buffer | object,
+    secretOrPrivateKey: null,
+    options?: SignOptions & { algorithm: "none" },
+): string;
```

Comment:
Not really; the logic in the JS is opposite

It says 

```
if (!secretOrPrivateKey && options.algorithm !== 'none') {
    return failure(new Error('secretOrPrivateKey must have a value'));
}
...continue...
```

the negative there is kind of annoying; it can be better expressed as 

```
if (!(secretOrPrivateKey != null || options.algorithm == 'none')) {
    return failure(new Error('secretOrPrivateKey must have a value'));
}
...continue...
```

or reverse

```
if (secretOrPrivateKey != null || options.algorithm == 'none') {
   ... continue...
   // return success
}
// return error
```

That is - either `options.algorithm == 'none'` is possible, or `secretOrPrivateKey` needs to be non-empty.

I can do a simple repro...

Reply:
```
var jwt = require("jsonwebtoken");

var s = jwt.sign({ foo: "bar" }, null, {
  algorithm: 'none',
})

console.log(s);
```

this works.

```
var jwt = require("jsonwebtoken");

var s = jwt.sign({ foo: "bar" }, null, {
  algorithm: 'HS256',
})

console.log(s);
```

this does not work and throws error on the line you linked.

Again, the "none" algorithm is kind of useless and shouldn't exist in JWT standard in the first place. However it does exist, and the jsonwebtoken supports it. And I am using the JWT library to test that my BE code doesn't actually accepts this "none" JWT token, so I want to create it in the test code.

Reply:
you're right, I was too focused on the `'none'` and missed that the comparison is `!==` not `===`